File: inzage-cms/backend/app/schemas/gst.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, HttpUrl

from app.schemas.ibron import IbronMinimalList
from app.schemas.oe import OeMinimalList


class Gst(BaseModel):
    gst_cd: int
    gst_upc: int
    versie_nr: int | None = None
    ext_lnk_aut: HttpUrl | None
    ibron_cd: int | None
    notitie: str | None
    oe_best: int
    oe_bron: int
    omschrijving: str
    conditie: str | None
    ts_mut: datetime
    user_nm: str


class GstMinimalList(BaseModel):
    gst_cd: int
    versie_nr: int | None
    omschrijving: str


class GstWithRelations(Gst):
    entity_ibron: IbronMinimalList | None
    entity_oe_best: OeMinimalList
    entity_oe_bron: OeMinimalList


class GstIn(BaseModel):
    ext_lnk_aut: str | None = None
    ibron_cd: int | None = None
    conditie: str | None = None
    notitie: str | None = None
    oe_best: int
    oe_bron: int
    omschrijving: str


GstWithRelations.model_rebuild()

File: inzage-cms/frontend/src/types/Gst.ts
export type Gst = {
  gst_cd: number
  gst_upc: number
  omschrijving: string
  oe_bron: number
  oe_best: number
  ibron: number
  ext_lnk_auth: string
  conditie: string
}

export type GstTypeTree = {
  gstt_cd: number
  gst_gstt_cd: number
  gstt_naam: string
  gstt_oms: string
  gstt_pp: string
}
